erpnext/patches/v7_1/set_total_amount_currency_in_je.py

```diff
@@ -0,0 +1,28 @@
+import frappe
+from frappe.utils import cint
+
+def execute():
+	for je in frappe.get_all("Journal Entry", fields=["name", "multi_currency"]):
+		if cint(je.multi_currency):
+			doc = frappe.get_doc("Journal Entry", je.name)
+			pay_to_recd_from, bank_account_currency, party_account_currency = None, None, None
+			
+			for d in doc.get('accounts'):
+				if d.party_type and d.party:
+					if not pay_to_recd_from:
+						pay_to_recd_from = frappe.db.get_value(d.party_type, d.party,
+							"customer_name" if d.party_type=="Customer" else "supplier_name")
+
+					party_account_currency = d.account_currency
+
+				elif frappe.db.get_value("Account", d.account, "account_type") in ["Bank", "Cash"]:
+					bank_account_currency = d.account_currency
+
+			if pay_to_recd_from:
+				if bank_account_currency:
+					frappe.db.set_value("Journal Entry", je.name, "currency", bank_account_currency,
```

Comment:
instead of if... else, you can write like:

`frappe.db.set_value("Journal Entry", je.name, "currency", bank_account_currency or party_account_currency, update_modified=False)`